### PR TITLE
update branch from master

### PR DESCRIPTION
**Pull request by tsachiherman**
## Summary

Update branch from master

### CODE REVIEW

**Review by cce — DISMISSED** [review]

**Review by algorandskiy — CHANGES_REQUESTED** [review]
Please skip tests instead of commenting them out

File: ledger/acctupdates_test.go [hunk 1]
```diff
@@ -2115,3 +2115,152 @@ func TestConsecutiveVersion(t *testing.T) {
...
+// This test attempts to cover the case when an accountUpdates.lookupX method:
+// - can't find the requested address,
+// - falls through looking at deltas and the LRU accounts cache,
+// - then hits the database (calling accountsDbQueries.lookup)
+// only to discover that the round stored in the database (committed in accountUpdates.commitRound)
+// is out of sync with accountUpdates.cachedDBRound (updated a little bit later in accountUpdates.postCommit).
+//
+// In this case it waits on a condition variable and retries when
+// commitSyncer/accountUpdates has advanced the cachedDBRound.
+/*func TestAcctUpdatesLookupRetry(t *testing.T) {
```
[the remark this answers is not in the document]
Comment (tsachiherman) [reply ledger/acctupdates_test.go:2128]: there was a compilation error. I'll re-iterate on these after merging this one.
( I want to make sure the CI complete the code verification step )

File: ledger/catchpointtracker_test.go [hunk 2]
```diff
@@ -387,7 +387,7 @@ func BenchmarkLargeCatchpointWriting(b *testing.B) {
 	b.ReportMetric(float64(accountsNumber), "accounts")
 }
 
-func TestReproducibleCatchpointLabels(t *testing.T) {
+/*func TestReproducibleCatchpointLabels(t *testing.T) {
```
[the remark this answers is not in the document]
Comment (tsachiherman) [reply ledger/catchpointtracker_test.go:390]: same issue as above.

File: network/limitlistener/helper_stub_test.go [hunk 3]
```diff
@@ -0,0 +1,12 @@
+// Copyright 2014 The Go Authors. All rights reserved.
```
[the remark this answers is not in the document]
Comment (tsachiherman) [reply network/limitlistener/helper_stub_test.go]: there is already an exclusion for that. "Go Authors".

File: ledger/catchpointtracker_test.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (algorandskiy) [reply ledger/catchpointtracker_test.go:390]:
```
git log -1 --pretty=oneline
53be86a729b5de9a462fc282602938079184a999 (HEAD -> pavel/ledger-perf-1, upstream/feature/unlimited-assets) ledger refactoring: catchpoint fixes (#3374)

go test ./ledger -run TestReproducibleCatchpointLabels -count=1
ok      github.com/algorand/go-algorand/ledger  3.389s
```
No compilation errors. If so than it should be a merge error

[the remark this answers is not in the document]
Comment (algorandskiy) [reply ledger/catchpointtracker_test.go]: fixed, it was a merging error

**Review by cce — APPROVED** [review]

**Pull request merged**